### PR TITLE
Adding changelog for v1.2.0

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds the changelog for v1.2.0. Merging this PR will kickoff the v1.2.0 release process.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @mlavacca @shaneutt @youngnick 
/hold

### CODE REVIEW

File: CHANGELOG/1.2-CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,619 @@
...
+resources. Unfortunately, this field is not automatically pruned. To check these
+values, you can run the following commands:
+
+```
+kubectl get crd grpcroutes.gateway.networking.k8s.io -ojsonpath="{.status.storedVersions}"
+kubectl get crd referencegrants.gateway.networking.k8s.io -ojsonpath="{.status.storedVersions}"
+```
+
+If either of these return a list that includes "v1alpha2", it means that we need
+to manually remove that version from `storedVersions`.
```
Comment (robscott) [inline CHANGELOG/1.2-CHANGELOG.md:46]: @mlavacca I ran out of time to finish this part - do you mind adding some instructions here? I think the gist would be:

1. A link to https://github.com/kubernetes-sigs/kube-storage-version-migrator
2. Some kind of simple script that makes a no-op update to all GRPCRoutes and ReferenceGrants in the cluster to make sure they've all been migrated to the latest storage version (likely simpler than running storage-version-migrator)
3. A kubectl patch command that updates `status.storedVersions` to drop v1alpha2

**Review by shaneutt — APPROVED** [review]
Keeping the hold for the moment, but LGTM.

/approve
/lgtm

**Review by shaneutt — COMMENTED** [review]
/lgtm

**Review by mlavacca — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**